crates/uv/src/commands/project/run.rs

```diff
@@ -717,6 +718,55 @@ pub(crate) async fn run(
         .as_ref()
         .map_or_else(|| &base_interpreter, |env| env.interpreter());
 
+    // Check if any run command is given.
+    // If not, print the available scripts for the current interpreter.
+    if let RunCommand::Empty = command {
+        writeln!(
+            printer.stdout(),
+            "Provide a command or script to invoke with `uv run <command>` or `uv run script.py`.\n"
+        )?;
+
+        let mut scripts = interpreter
+            .scripts()
+            .read_dir()
+            .ok()
+            .into_iter()
+            .flatten()
+            .filter_map(|entry| match entry {
+                Ok(entry) => Some(entry),
+                Err(err) => {
+                    debug!("Failed to read entry: {}", err);
+                    None
+                }
+            })
+            .filter(|entry| {
+                entry
+                    .file_type()
+                    .is_ok_and(|file_type| file_type.is_file() || file_type.is_symlink())
+            })
+            .map(|entry| entry.path())
+            .filter(|path| is_executable(path))
+            .map(|path| {
+                path.file_name()
+                    .unwrap_or_default()
+                    .to_string_lossy()
+                    .to_string()
+            })
+            .collect_vec();
+        scripts.sort();
+
+        if !scripts.is_empty() {
+            writeln!(printer.stdout(), "The following scripts are available:\n")?;
+            writeln!(printer.stdout(), "{}", scripts.join("\n"))?;
+        }
+        writeln!(
+            printer.stdout(),
+            "\nSee `uv run --help` for more information."
+        )?;
+
+        return Ok(ExitStatus::Success);
```

Comment:
Should we exit with a non-zero exit code? I'm not sure but I think so?

Reply:
Let me check what we do for the `uv tool run`. But it makes sense to return an unsuccessful exit code. I favor consistency, so whatever we decide should be the same all over the board. What do you think?

Reply:
Showing the help menu exits with code 2 right now — I think we should be consistent with that.

Reply:
👍 Done. I've also updated the https://github.com/astral-sh/uv/pull/7641